python/cudf_polars/cudf_polars/dsl/translate.py

```diff
@@ -0,0 +1,407 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Translate polars IR representation to ours."""
+
+from __future__ import annotations
+
+from contextlib import AbstractContextManager, nullcontext
+from functools import singledispatch
+from typing import Any
+
+from polars.polars import _expr_nodes as pl_expr, _ir_nodes as pl_ir
```

Comment:
Is `_ir_nodes` eventually going to be made public? Do we need to push for that?

Reply:
No, I do not believe so. I also don't think we need to push for it.